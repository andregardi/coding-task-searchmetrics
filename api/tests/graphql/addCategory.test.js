const { createTestClient } = require('apollo-server-testing');
const apolloServer = require('../../src/graphql/apolloServer');
const datamuseClient = require('../../src/services/datamuse/client');
const addCategoryMutations = require('./__mutations__/addCategory');

jest.mock('../../src/services/datamuse/client', () => ({
  get: jest.fn(),
}));

describe('addCategory', () => {
  describe('on success', () => {
    it('it should match the snapshot', async () => {
      const mockedKeywords = 'mockedKeywords';
      datamuseClient.get.mockResolvedValueOnce({ data: [mockedKeywords] });

      const { mutate } = createTestClient(apolloServer);
      const res = await mutate({
        mutation: addCategoryMutations.successMutation,
      });

      expect(res).toMatchSnapshot();
    });
  });

  describe('if the category already exists', () => {
    it('should return error', async () => {
      const { mutate } = createTestClient(apolloServer);
      const { errors } = await mutate({
        mutation: addCategoryMutations.carsAlreadyExist,
      });

      expect(errors[0].message).toMatch(/Category cars already exists/);
    });
  });
});
