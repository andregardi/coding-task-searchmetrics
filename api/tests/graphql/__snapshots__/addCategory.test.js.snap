// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addCategory on success it should match the snapshot 1`] = `
Object {
  "data": Object {
    "addCategory": Object {
      "keywords": Array [
        null,
      ],
      "name": "newCategory",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
